### PR TITLE
[#160368] Facility card view

### PR DESCRIPTION
**Pull request by jossim**
# Release Notes

This adds the `facility_tile_list` feature flag. When set to `true`, it will turn the list of facilities on the home page into a tile grid, where each facility can have it's on image.

The images are handled by `ActiveStorage` or `Paperclip`, depending on the `active_storage` feature flag. If using `Paperclip`, a migration is required, which is outlined in the CHANGELOG.

# Screenshot
Screen Shot 2022-10-14 at 6 20 38 PM

# Additional context

This PR is based on this older PR from UConn https://github.com/tablexi/nucore-open/pull/1803

### CODE REVIEW

**Comment by jossim** [comment]
To process images (e.g. resize them), looks like the `image_processing` gem is required for `ActiveStorage`. It [requires either libvips or ImageMagick](https://guides.rubyonrails.org/active_storage_overview.html#requirements) to be installed on the server

File: app/models/facility.rb [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 class Facility < ApplicationRecord
 
+  # Migration is required if Paperclip is being used.
+  # The image is used in the UI if the facility_tile_list feature flag is enabled
+  include DownloadableFiles::Image
```
Comment (giladshanan) [inline app/models/facility.rb:7]: I think this needs a feature flag, right?  Maybe the code comment would be more at home in the Image concern?

File: app/views/facilities/_facility_fields.html.haml [hunk 2]
```diff
@@ -2,6 +2,9 @@
   = f.input :name, hint: text(".hints.name")
   = f.input :abbreviation, hint: text(".hints.abbreviation")
   = f.input :url_name, hint: facility_url(f.object.url_name || "url-name"), as: :string
+  - if SettingsHelper.feature_on?(:facility_tile_list)
+    = f.input :file, label: "Image"
```
Comment (giladshanan) [inline app/views/facilities/_facility_fields.html.haml:6]: i would expect this to be:
```suggestion
    = f.input :file_field, :file, label: "Image"
```
... but if its working then maybe I'm wrong :)

File: config/settings.yml [hunk 3]
```diff
@@ -126,7 +127,7 @@ feature:
   kiosk_view: true
   bypass_kiosk_auth: false
   revenue_account_editable: false
-  active_storage: false
+  active_storage: true
```
Comment (giladshanan) [inline config/settings.yml:130]: I would rather do this in a separate PR

**Review by giladshanan — COMMENTED** [review]
Looks like a great start!  Left a few coments

**Comment by jossim** [comment]
It'll probably be a good idea to consider image processing to ensure the images people upload aren't too large. This will make the infrastructure more complicated however.

File: spec/system/facility_list_tile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      expect(page).to have_selector(".tile-image")
+    end
+
+    context "image manpulation" do
+      before do
+        login_as admin_user
+        visit edit_facility_path(facility)
+      end
+
+      context "with image attached" do
```
Comment (giladshanan) [inline spec/system/facility_list_tile_spec.rb:28]: I'd suggest making with/without an attached image the top level context.  Fine to do this in a follow up PR

**Review by giladshanan — APPROVED** [review]
